pkg/middleware/request_tracing.go

```diff
@@ -0,0 +1,35 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+
+	"gopkg.in/macaron.v1"
+)
+
+func RequestTracing() macaron.Handler {
+	return func(res http.ResponseWriter, req *http.Request, c *macaron.Context) {
+		rw := res.(macaron.ResponseWriter)
+
+		var span opentracing.Span
+		tracer := opentracing.GlobalTracer()
+		wireContext, _ := tracer.Extract(opentracing.HTTPHeaders, opentracing.HTTPHeadersCarrier(req.Header))
+		spanName := fmt.Sprintf("HTTP %s %s", req.Method, req.URL.Path)
```

Comment:
using url.path for span is usually not recommended, because REST conventions encourage high cardinality, e.g. `/api/customer/{uuid}`

Reply:
Would it make sense to use the url routing path instead?

ex 
```
/api/customer/{uuid}
```
instead of 
```
/api/customer/6574949
```

Reply:
what's wrong with high cardinality? i thought OT/jaeger were designed for high cardinality logs, tags etc.

Reply:
@bergquist yes, if you get access to the route it's perfect as the endpoint name

@Dieterbe high cardinality is fine for tags and logs, but not for the operation name. First, operation name is a dropdown in the UI, you don't want unlimited domain there. But more importantly, if you don't have canonical names for your endpoints, you cannot do any aggregations of traces, or even emit RPC metrics.

Reply:
aha makes sense